modules/sdk-coin-avaxp/src/lib/delegatorTxBuilder.ts

```diff
@@ -243,41 +233,73 @@ export class DelegatorTxBuilder extends TransactionBuilder {
    * 0: user key, 1: recovery key, 2: hsm key
    * @protected
    */
-  protected createInputOutput(): { inputs: TransferableInput[]; outputs: TransferableOutput[] } {
+  protected createInputOutput(): {
+    inputs: TransferableInput[];
+    outputs: TransferableOutput[];
+    credentials: Credential[];
+  } {
     const inputs: TransferableInput[] = [];
     const outputs: TransferableOutput[] = [];
     const addresses = this.transaction._fromAddresses.map((b) =>
       utils.addressToString(this.transaction._network.hrp, this.transaction._network.alias, b)
     );
     let total: BN = new BN(0);
     const totalTarget = this._stakeAmount.clone().add(this.transaction._txFee);
+    const credentials: Credential[] = [];
+
+    this.transaction._utxos.forEach((outputs) => {
+      if (!outputs.addressesIndex || outputs.addressesIndex.length == 0) {
+        outputs.addressesIndex = addresses.map((a) => outputs.addresses.indexOf(a));
+      }
+    });
+
     this.transaction._utxos
       .filter(
-        (output) => output.threshold === this.transaction._threshold && utils.includeIn(addresses, output.addresses)
+        (output) =>
+          output.threshold === this.transaction._threshold &&
+          (output.addressesIndex?.length == 3 || !output.addressesIndex?.includes(-1))
       )
       .forEach((output, i) => {
         if (output.outputID === 7 && total.lte(totalTarget)) {
```

Comment:
nit: what does this magic number mean?

Reply:
I copy from coin-sandbox. 

@jicli @sskim16 Could you help with this?

Reply:
there too!
https://github.com/BitGo/BitGoJS/blob/9d6de166ed47f6cf1491b694c79115454fee57e1/modules/sdk-coin-avaxp/src/lib/delegatorTxBuilder.ts#L232

Es el mismo en ambos casos.